### DISCUSSION TITLE
Request to be more explicit on the docker install.

### DISCUSSION THREAD

**Discussion by michaeltaylorj2ee** [Taxel/PlexTraktSync#769] [Ideas]
I was having issues installing this to my docker instance and thought that there might be a better way to describe how to install on docker. It is not very explicit how you get to the setup by leaving out the watch command. It also might be helpful to indicate that it is best to do the initial installation via the command line and then eventually managing the installation with something like Portainer.

That being said, if anyone else has trouble, hopefully this post will help you if you encounter the same issues.

I was getting this error initially:

    Plex connection error: HTTPConnectionPool(host='localhost', port=32400): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xb56836e8>: Failed to establish a new connection: [Errno 111] Connection refused')), fallback url None didn't respond either.    

I was having trouble getting into the shell once the container started. I was trying to use portainer to install, but ended up using SSH to install. Not sure if there is a way to use stack in portainer and immediately go to the shell, but this worked.

My tech is:

* Raspberry Pi 4
* Debian with OMV 5
* Docker with Portainer
* External HDDs

Login to your system with SSH and create a docker-compose.yml file. Here is mine.

    version: "2.1"
    services:
      plextraktsync:
        image: ghcr.io/taxel/plextraktsync
        container_name: plextraktsync
        volumes:
          - /srv/dev-disk-by-uuid-YOUR-DISK/data/appdata/configs/plextraktsync/config:/app/config

Then run this in SSH:

     docker-compose run --rm plextraktsync
 

Then complete the install with the command line and you will get your env file created.

Then run this in SSH

     docker-compose run --rm plextraktsync watch

My plan now is to remove this initial container and replace it with my stack.

**Comment by glensc** [comment]
To submit a documentation update, send a pull request.

**Comment by agneevX** [comment]
This is a known limitation, as you have to start the initial setup process while attached to a tty. 

The docker-compose.yml in the repo assumes that you already have it setup whereas the README includes instruction for the setup process. 

I plan to submit a PR sometime soon to better document the process. 

I'm not quite sure about creating containers with Portainer, but there should be an option for it to attach to a shell. This would be fair to document.
